src/neptune/internal/backends/hosted_file_operations.py

```diff
@@ -482,14 +479,13 @@ def _store_response_as_file(
         target_file = destination
 
     total_size = int(response.headers.get("content-length", 0))
-    pre_download_hook(total_size)
-    with response:
-        with open(target_file, "wb") as f:
-            for chunk in response.iter_content(chunk_size=1024 * 1024):
-                if chunk:
-                    f.write(chunk)
-                download_iter_hook(len(chunk) if chunk else 0)
-    post_download_hook()
+    with construct_progress_bar(progress_bar, "Fetching fileset...") as bar:
+        with response:
+            with open(target_file, "wb") as f:
+                for chunk in response.iter_content(chunk_size=1024 * 1024):
+                    if chunk:
+                        f.write(chunk)
+                        bar.update(by=len(chunk), total=total_size)
```

Comment:
its getting a bit nested. There is cool python trick, you can actually use `with` with multiple context managers with only one indent :))

```
    total_size = int(response.headers.get("content-length", 0))
    with (
        construct_progress_bar(progress_bar, "Fetching file...") as bar,
        response,
        open(target_file, "wb") as f
    ):
        for chunk in response.iter_content(chunk_size=1024 * 1024):
            if chunk:
                f.write(chunk)
                bar.update(by=len(chunk), total=total_size)
```

its added in python 3.10 so not sure if we can use it tho :(

but there are other ways around it in pre 3.10 versions like

```
with A() as a, \
     B() as b, \
     C() as c:
    doSomething(a,b,c)
```

more info: https://stackoverflow.com/questions/893333/multiple-variables-in-a-with-statement
python 3.10 doc: https://docs.python.org/3/whatsnew/3.10.html#parenthesized-context-managers

Reply:
Thought of it too, but unfortunately, it is not compatible with lower python versions, so we cannot support that. I can leave a TODO though to update it once we have 3.10 as the minimum version, what do you think?
Also, a workaround here could be to use `try-finally` syntax, but I'm not certain if we need that so much

Reply:
yep I think moving from python 3.7 to newer versions is crucial as we will keep running into issues like that.

1-st seems good, I'm not big fan of `try-finally` approach (its only nesting at the end).
How about [ExitStack](https://docs.python.org/3/library/contextlib.html#contextlib.ExitStack) have you seen that? Its supported since python 3.3

Reply:
Didn't know that. Looks promising :)